lbry/lbry/dht/node.py

```diff
@@ -121,56 +126,55 @@ def stop(self) -> None:
+                        (address, udp_port) != (self.protocol.external_ip, self.protocol.udp_port):
+                    peer_addresses.append((node_id, address, udp_port, tcp_port))
+            return [make_kademlia_peer(*peer_address) for peer_address in peer_addresses]
+
+        def set_joined():
+            self.joined.set()
+            log.info(
+                "joined dht, %i peers known in %i buckets", len(self.protocol.routing_table.get_peers()),
+                self.protocol.routing_table.buckets_with_contacts()
+            )
+
         if not self.listening_port:
             await self.start_listening(interface)
         self.protocol.ping_queue.start()
         self._refresh_task = self.loop.create_task(self.refresh_node())
 
-        # resolve the known node urls
-        known_node_addresses = []
-        url_to_addr = {}
-
-        if known_node_urls:
-            for host, port in known_node_urls:
-                address = await resolve_host(host, port, proto='udp')
-                if (address, port) not in known_node_addresses and\
-                        (address, port) != (self.protocol.external_ip, self.protocol.udp_port):
-                    known_node_addresses.append((address, port))
-                    url_to_addr[address] = host
-
-        if known_node_addresses:
-            peers = [
-                make_kademlia_peer(None, address, port)
-                for (address, port) in known_node_addresses
-            ]
-            while True:
-                if not self.protocol.routing_table.get_peers():
-                    if self.joined.is_set():
-                        self.joined.clear()
-                    self.protocol.peer_manager.reset()
-                    self.protocol.ping_queue.enqueue_maybe_ping(*peers, delay=0.0)
-                    peers.extend(await self.peer_search(self.protocol.node_id, shortlist=peers, count=32))
-                    if self.protocol.routing_table.get_peers():
-                        self.joined.set()
-                        log.info(
-                            "Joined DHT, %i peers known in %i buckets", len(self.protocol.routing_table.get_peers()),
-                            self.protocol.routing_table.buckets_with_contacts())
-                    else:
-                        continue
-                await asyncio.sleep(1, loop=self.loop)
-
-        log.info("Joined DHT, %i peers known in %i buckets", len(self.protocol.routing_table.get_peers()),
-                 self.protocol.routing_table.buckets_with_contacts())
-        self.joined.set()
-
-    def start(self, interface: str, known_node_urls: typing.List[typing.Tuple[str, int]]):
-        self._join_task = self.loop.create_task(
-            self.join_network(
-                interface=interface, known_node_urls=known_node_urls
-            )
-        )
+        restored_peers = peers_from_urls(await self._storage.get_peers()) if self._storage else []
+
+        fixed_peers = peers_from_urls([
```

Comment:
this should be in the loop to allow reconnecting from the seeds if the internet connection drops/comes back. if `seed_peers` is empty, it can be extended with the fixed seeds.

Reply:
Currently, the `seed_peers` are already changed if the search fails (and can easily extend), but doing it conditionally within the loop potentially saves DNS lookup, so agreed.